src/main/java/uk/gov/hmcts/ccd/data/definition/CachedCaseDefinitionRepository.java

```diff
@@ -4,90 +4,79 @@
+import org.springframework.cache.annotation.Cacheable;
 import org.springframework.stereotype.Service;
-import org.springframework.web.context.annotation.RequestScope;
 import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
 import uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition;
 import uk.gov.hmcts.ccd.domain.model.definition.JurisdictionDefinition;
 import uk.gov.hmcts.ccd.domain.model.definition.UserRole;
 
+import javax.inject.Singleton;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
-import static com.google.common.collect.Maps.newHashMap;
-
 @Service
+@Singleton
 @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
-@RequestScope
 // TODO: Make this repository return copies of the maps https://tools.hmcts.net/jira/browse/RDM-1459
 public class CachedCaseDefinitionRepository implements CaseDefinitionRepository {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(CachedCaseDefinitionRepository.class);
-
     public static final String QUALIFIER = "cached";
+    private static final Logger LOGGER = LoggerFactory.getLogger(CachedCaseDefinitionRepository.class);
 
     private final CaseDefinitionRepository caseDefinitionRepository;
-    private final Map<String, List<CaseTypeDefinition>> caseTypesForJurisdictions = newHashMap();
-    private final Map<String, CaseTypeDefinitionVersion> versions = newHashMap();
-    private final Map<String, UserRole> userRoleClassifications = newHashMap();
-    private final Map<String, List<FieldTypeDefinition>> baseTypes = newHashMap();
+
+    @Autowired
+    @SuppressWarnings("checkstyle:MemberName")
+    private CachedCaseDefinitionRepository _this;
 
     @Autowired
     public CachedCaseDefinitionRepository(@Qualifier(DefaultCaseDefinitionRepository.QUALIFIER)
-                                                  CaseDefinitionRepository caseDefinitionRepository) {
+                                              final CaseDefinitionRepository caseDefinitionRepository) {
         this.caseDefinitionRepository = caseDefinitionRepository;
     }
 
     @Override
+    @Cacheable("caseTypesForJurisdictionCache")
     public List<CaseTypeDefinition> getCaseTypesForJurisdiction(final String jurisdictionId) {
-        return caseTypesForJurisdictions.computeIfAbsent(jurisdictionId,
-                                                         caseDefinitionRepository::getCaseTypesForJurisdiction);
+        return caseDefinitionRepository.getCaseTypesForJurisdiction(jurisdictionId);
     }
 
     @Override
     public CaseTypeDefinition getCaseType(final String caseTypeId) {
-        CaseTypeDefinitionVersion latestVersion = this.getLatestVersion(caseTypeId);
-        return caseDefinitionRepository.getCaseType(latestVersion.getVersion(), caseTypeId);
+        CaseTypeDefinitionVersion latestVersion = _this.getLatestVersion(caseTypeId);
+        return _this.getCaseType(latestVersion.getVersion(), caseTypeId);
```

Comment:
it invokes normal execution rather than cached version. Spring proxy limitation.. 
google it about calling  spring cache method from another method in the same class.